### PR TITLE
Stampie 1.0 compatibility

### PR DESCRIPTION
**Pull request by fbourigault**
This add stampie/stampie 1.0 compatibility. This need stof/StampieExtra#4.

I removed all adapter related stuff and added a the `hb_stampie.http_client` configuration option which is a service id that must be a `Http\Client\HttpClient`. The default value is `hb_stampie.http_client`. This value is an alias to `httplug.client.default` when `HttplugBundle` is enabled and a definition found using discovery when not.

### CODE REVIEW

**Comment by fbourigault** [comment]
@Nyholm this looks ready to me. What do you think about the optional HttplugBundle integration?

File: DependencyInjection/Configuration.php [hunk 1]
```diff
@@ -38,7 +38,7 @@ public function getConfigTreeBuilder()
             ->children()
                 ->scalarNode('mailer')->isRequired()->end()
                 ->scalarNode('server_token')->isRequired()->end()
-                ->scalarNode('adapter')->isRequired()->end()
+                ->scalarNode('http_client')->defaultValue('hb_stampie.http_client')->end()
```
Comment (Nyholm) [inline DependencyInjection/Configuration.php]: Let’s default it to “httplug.client”

File: DependencyInjection/HBStampieExtension.php [hunk 2]
```diff
@@ -32,12 +32,11 @@ public function load(array $configs, ContainerBuilder $container)
         $processor = new Processor();
         $config    = $processor->processConfiguration(new Configuration($container->getParameter('kernel.debug')), $configs);
 
-        $adapterServiceId = sprintf('hb_stampie.adapter.%s', $config['adapter']);
-        $mailerServiceId  = sprintf("hb_stampie.mailer.%s", $config['mailer']);
 
-        if (!$container->has($adapterServiceId)) {
-            throw new \InvalidArgumentException(sprintf('Invalid adapter "%s" specified', $config['adapter']));
-        }
+        $bundles = $container->getParameter('kernel.bundles');
```
Comment (Nyholm) [inline DependencyInjection/HBStampieExtension.php]: Let’s not do this hack. We should only assume that the string the user provided in the config is a proper service of a HTTPlugCliet

**Comment by fbourigault** [comment]
What would you write as `php-http/httplug-bundle` suggest hint?

**Comment by Nyholm** [comment]
"To make it easier to configure your HTTP clients"

**Comment by fbourigault** [comment]
Do we handle documentation update in this PR or in a separate one?

**Pull request merged**